lib/AST/GenerateAST.cpp

```diff
@@ -203,12 +204,114 @@ clang::Expr *GenerateAST::GetOrCreateReachingCond(llvm::BasicBlock *block) {
   return cond;
 }
 
+class BlockVisitor : public llvm::InstVisitor<BlockVisitor, clang::Stmt *> {
```

Comment:
What's the exact purpose of this class? How does it relate to `IRToASTVisitor` and how is it different?